src/core/libmaven/PeakDetector.cpp

```diff
@@ -415,6 +504,12 @@ void PeakDetector::processSlices(vector<mzSlice*> &slices, string setName)
                                      mavenParameters->limitGroupCount));
         }
     }
+
+    // finally, we would like to remove adducts for which no parent ions were
+    // detected or if parent ion's RT is too different from adduct's.
+    sendBoostSignal("Filtering out false adducts…", 0, 0);
+    GroupFiltering groupFilter(mavenParameters);
+    groupFilter.filterAdducts(mavenParameters->allgroups);
```

Comment:
As discussed offline, please change the adduct detection flow to mimic how isotopes are pulled in El-MAVEN. It will reduce the amount of time spent detecting and then deleting adduct groups where the parent is not found or the RT does not match.

Reply:
> Done. I've kept the `filterAdducts` method. I think its still required. But adduct slices are now only created from any detected peak-groups.

Please ignore the original message. It was meant as a reply to this comment.